java-checks/src/main/java/org/sonar/java/checks/spring/SpelExpressionCheck.java

```diff
@@ -0,0 +1,246 @@
+import org.sonar.plugins.java.api.tree.MethodTree;
+import org.sonar.plugins.java.api.tree.Tree;
+import org.sonar.plugins.java.api.tree.VariableTree;
+import org.springframework.expression.ParseException;
+import org.springframework.expression.spel.standard.SpelExpressionParser;
+
+@Rule(key = "S6857")
+public class SpelExpressionCheck extends IssuableSubscriptionVisitor {
+
+  private static final String SPRING_PREFIX = "org.springframework";
+
+  private static final Pattern PROPERTY_PLACEHOLDER_PATTERN = Pattern.compile(
+    "[a-zA-Z_]\\w*+(\\[\\d++])*+(\\.[a-zA-Z_]\\w*+(\\[\\d++])*+)*+"
+  );
+
+  public List<Tree.Kind> nodesToVisit() {
+    return List.of(Tree.Kind.CLASS, Tree.Kind.INTERFACE, Tree.Kind.ANNOTATION_TYPE);
+  }
+
+  @Override
+  public void visitNode(Tree tree) {
+    getClassAndMemberAnnotations((ClassTree) tree)
+      .filter(SpelExpressionCheck::isSpringAnnotation)
+      .forEach(this::checkSpringAnnotationArguments);
+  }
+
+  private static Stream<AnnotationTree> getClassAndMemberAnnotations(ClassTree cls) {
+    return Stream.concat(
+      Stream.of(cls.modifiers().annotations()),
+      cls.members().stream().map(SpelExpressionCheck::getMemberAnnotations)
+    ).flatMap(Collection::stream);
+  }
+
+  private static List<AnnotationTree> getMemberAnnotations(Tree member) {
+    if (member.is(Tree.Kind.METHOD)) {
+      return ((MethodTree) member).modifiers().annotations();
+    } else if (member.is(Tree.Kind.VARIABLE)) {
+      return ((VariableTree) member).modifiers().annotations();
+    } else {
+      return Collections.emptyList();
+    }
+  }
+
+  private static boolean isSpringAnnotation(AnnotationTree annotation) {
+    return annotation.symbolType().fullyQualifiedName().startsWith(SPRING_PREFIX);
+  }
+
+  private void checkSpringAnnotationArguments(AnnotationTree annotation) {
+    annotation.arguments().stream()
+      .filter(argument -> argument.is(Tree.Kind.STRING_LITERAL))
+      .forEach(argument -> checkSpringAnnotationStringArgument((LiteralTree) argument));
+  }
+
+  private void checkSpringAnnotationStringArgument(LiteralTree argument) {
+    try {
+      var argValue = argument.value();
+      var argContents = argValue.substring(1, argValue.length() - 1);
+      parseStringContents(argContents);
+    } catch (SyntaxError e) {
+      reportIssue(argument, e);
+    }
+  }
+
+  private void reportIssue(LiteralTree stringToken, SyntaxError error) {
+    var tokenStart = Position.startOf(stringToken);
+    var textSpan = new AnalyzerMessage.TextSpan(
+      tokenStart.line(),
+      tokenStart.columnOffset() + error.rangeStart,
+      tokenStart.line(),
+      tokenStart.columnOffset() + error.rangeEnd
+    );
+
+    var analyzerMessage = new AnalyzerMessage(this, context.getInputFile(), textSpan, error.getMessage(), 0);
+    ((DefaultJavaFileScannerContext) context).reportIssue(analyzerMessage);
+  }
+
+  private static void parseStringContents(String content) {
```

Comment:
We should add the exception to the signature since the this is the last level it can bubble up from
```suggestion
  private static void parseStringContents(String content) throws SyntaxError {
```